diff --git a/initramfs/init b/initramfs/init
index 29f56b7..9c97d31 100755
--- a/initramfs/init
+++ b/initramfs/init
@@ -102,7 +102,7 @@ find_and_mount_media() {
 	echo -e " ${BOLD}${BLUE}*${NORM} Populating root filesystem..."
 	mkdir -p /newroot
 	echo -e -n "   ${BOLD}${BLUE}*${NORM} Mounting new root filesystem... "
-	mount -t tmpfs tmpfs /newroot -o size=761m
+	mount -t tmpfs tmpfs /newroot -o size=1002m
 	checkReturn
 	echo -e -n "   ${BOLD}${BLUE}*${NORM} Copying files from rootfs.tar.xz... "
 	tar -xJf ${MEDIAMNT}/${PREFIX}/rootfs.tar.xz -C /newroot
@@ -265,7 +265,21 @@ then
 		echo -n ", "
 	}
 
-	MODORDER="fs cdrom ata scsi usb firewire"
+	# Apple
+	load_apple_modules() {
+		echo -e -n "${BOLD}${BLUE}Apple${NORM}"
+		for mod in apple-ibridge apple-bce apple-ib-als apple-ib-tb
+		do
+			if [ ! -z "$DEBUG" ]
+			then
+				echo -n " $mod"
+			fi
+			modprobe $mod
+		done
+		echo -n ", "
+	}
+
+	MODORDER="fs cdrom ata scsi usb firewire apple"
 
 	for MODS in $MODORDER
 	do
