diff --git a/pkgmk.conf b/pkgmk.conf
index 5bf3fc2..d9f7630 100644
--- a/pkgmk.conf
+++ b/pkgmk.conf
@@ -28,8 +28,8 @@ esac
 # PKGMK_PACKAGE_DIR="/usr/ports/packages"
 # PKGMK_WORK_DIR="/usr/ports/work/$name"
 # PKGMK_DOWNLOAD="no"
-# PKGMK_IGNORE_FOOTPRINT="no"
-# PKGMK_IGNORE_NEW="no"
+PKGMK_IGNORE_FOOTPRINT="yes"
+PKGMK_IGNORE_NEW="yes"
 # PKGMK_NO_STRIP="no"
 # PKGMK_WGET_OPTS=""
 PKGMK_COMPRESSION_MODE="xz"
